src/julia_internal.h

```diff
@@ -170,79 +176,60 @@ static const int jl_gc_sizeclasses[JL_GC_N_POOLS] = {
 //    64,   32,  160,   64,   16,   64,  112,  128, bytes lost
 };
 
-STATIC_INLINE int jl_gc_alignment(size_t sz)
+STATIC_INLINE int JL_CONST_FUNC jl_gc_alignsz(size_t sz, size_t alignment)
 {
-    if (sz == 0)
-        return sizeof(void*);
-#ifdef _P64
-    (void)sz;
-    return 16;
-#elif defined(_CPU_ARM_) || defined(_CPU_PPC_) || defined(_CPU_X86_)
-    return sz <= 4 ? 8 : 16;
-#else
-    // szclass 8
-    if (sz <= 4)
-        return 8;
-    // szclass 12
-    if (sz <= 8)
-        return 4;
-    // szclass 16+
-    return 16;
-#endif
+    // The pools are aligned with JL_HEAP_ALIGNMENT and no bigger alignment is possible.
+    assert(alignment <= JL_HEAP_ALIGNMENT);
+    // Pools with the correct alignment will have an object size that
+    // is a multiple of the alignment. As an example an allocation with
+    // sz of 48 and an alignment of 32 will need to be in pool of size 64.
+    // An alignment of 0 or 1 means unaligned and we can use sz directly.
+    if (alignment != 0 && alignment != 1 && alignment != sz)
+        sz = ((sz / alignment) + 1) * alignment;
+    return sz;
 }
-JL_DLLEXPORT int jl_alignment(size_t sz);
 
-STATIC_INLINE int JL_CONST_FUNC jl_gc_szclass(size_t sz)
+// Use jl_gc_alignsz to obtain the right sz.
+STATIC_INLINE int JL_CONST_FUNC jl_gc_szclass(size_t sz, size_t alignment)
 {
-#ifdef _P64
-    if (sz <=    8)
-        return 0;
-    const int N = 0;
-#elif defined(_CPU_ARM_) || defined(_CPU_PPC_) || defined(_CPU_X86_)
-    if (sz <=    8)
-        return (sz + 3) / 4 - 1;
-    const int N = 1;
-#else
-    if (sz <=   12)
-        return (sz + 3) / 4 - 1;
-    const int N = 2;
-#endif
-    if (sz <=  256)
-        return (sz + 15) / 16 + N;
-    if (sz <=  496)
-        return 16 - 16376 / 4 / LLT_ALIGN(sz, 16 * 4) + 16 + N;
-    if (sz <= 1008)
-        return 16 - 16376 / 2 / LLT_ALIGN(sz, 16 * 2) + 24 + N;
-    return     16 - 16376 / 1 / LLT_ALIGN(sz, 16 * 1) + 32 + N;
+    size_t klass = 0;
+    while (klass < JL_GC_N_POOLS) {
```

Comment:
Yeah I know I wanted a fast way to get a fast way of checking correctness before going back to the jump table.